### PR TITLE
Azure Firewall NAT Rule Collection

### PR DESCRIPTION
**Pull request by fanymanea**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json [hunk 1]
```diff
@@ -473,6 +473,128 @@
          },
          "description":"Properties of an application rule."
       },
+      "AzureFirewallNatRuleCollectionPropertiesFormat": {
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json]: nit: Please rename this to `AzureFirewallNatRuleCollectionProperties` for brevity

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json [hunk 2]
```diff
@@ -473,6 +473,128 @@
...
+               "minimum":100,
+               "exclusiveMinimum":false,
+               "description":"Priority of the NAT rule collection resource."
+            },
+            "action":{  
+               "$ref":"#/definitions/AzureFirewallNatRCAction",
+               "description":"The action type of a NAT rule collection"
+            },
+            "rules":{  
+               "type":"array",
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json]: Please check if any of these properties are readonly and mark them accordingly

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json [hunk 3]
```diff
@@ -473,6 +473,128 @@
...
+         },
+         "description":"Properties of the NAT rule collection."
+      },
+      "AzureFirewallNatRuleCollection":{  
+         "properties":{  
+            "properties":{  
+               "x-ms-client-flatten":true,
+               "$ref":"#/definitions/AzureFirewallNatRuleCollectionPropertiesFormat"
+            },
+            "name":{
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json]: Please check if this is readonly

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json [hunk 4]
```diff
@@ -473,6 +473,128 @@
...
+               }
+            },
+            "destinationAddresses":{  
+               "type":"array",
+               "description":"List of destination IP addresses for this rule.",
+               "items":{  
+                  "type":"string"
+               }
+            },
+            "destinationPorts":{
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json]: nit: ports generally have a certain format, if this property is not readonly, it's a good idea to define a regex pattern for it
This applies to all other port/address types

**Review by dsgouda — CHANGES_REQUESTED** [review]
looks good for the most part, left a few comments

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json (hunk 2 above)
Comment (fanymanea) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json]: I marked etag. Any of the rest can be updated by user.

**Review by dsgouda — APPROVED** [review]
LGTM subject to CIs passing

**Comment by dsgouda** [comment]
@veronicagg Could you help with `NRP-Network Testing` job here? What does it test?

**Comment by veronicagg** [comment]
@dsgouda this job is specific to Network RP to check on generation of SDKs and validation of PR and CLI commands I believe, @shawns1 helped create it with guidance from @MikhailTryakhov, who has more context on it.

**Comment by number213** [comment]
@dsgouda this job tests if API version is live and tries to build & test PS/CLI with SDKs generated from PR's Swagger. Given that we update API version, there are no recordings with 2018-08-01 which results in failed check.
@veronicagg is there a way to mark this check as optional to avoid confusion?

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**